rfcs/0001-authentication/text.md

```diff
@@ -0,0 +1,383 @@
+  , and directs the client to save the JWT in an httpOnly cookie on the client.
+- The JWT proves the identity of the client. Subsequent requests from the client to the
+  cellxgene server will include the cookie with the JWT. When the cellxgene client re-requests
+  the configuration through the dataset’s “config” endpoint (&lt;server>/d/&lt;datasetname>/api
+  /v0.2/config), the configuration will now indicate that the user is authenticated. The
+  cellxgene client can also have access to the user’s name, which it may use when displaying the log information.
+  1. If the ID token is invalid, then the cellxgene server can re-initiate the authentication
+     process by going back to step [1]
+  2. The server will verify the signature of the token to ensure that it was signed by the
+     sender and not altered.
+  3. If the ID token is expired, then the server will obtain a new ID token automatically
+     using the Refresh token. It will then save the new ID token in the cookie.
+- When the dataset’s “schema” (&lt;server>/d/&lt;datasetname>/api/v0.2/schema) endpoint is
+  accessed from the cellxgene client, it will now include information about the user’s previously
+  created user annotations (if any).
+
+#### Logout flow
+
+The Logout flow is much simpler.
+
+- A request is sent to the to the “/logout” endpoint in cellxgene
+- cellxgene redirects back to the current page, with a directive to delete the cookie containing the JWT.
+
+#### Application Endpoints
+
+To support the Auth0 authentication, following endpoints are needed by cellxgene. These endpoints were referenced in the previous sections, and defined here in more detail.
+
+Note: If authentication is not enabled, then these endpoints will not be present (e.g. they would all return 404 errors).
+
+- /login : this endpoint is used to initiate the login process. The client passes in the
+  location to the current dataset in the request. The server then initiated the authentication
+  request by redirecting the browser to the Auth0 universal login page. The server passes in the
+  location of the dataset as a state variable in the request, so the browser can be redirected
+  there after login. This is step 1 in the diagram. /login is requested when the user clicks on
+  the “login” button.
+- /logout : this endpoint is used to clear the authentication credentials. The cookie
+  containing the ID token is removed. /logout is requested when the user clicks on the “logout” button.
+- /oauth2/callback: this endpoint is accessed during step 3 of the diagram. Auth0 redirects
+  the user to this endpoint, and the request contains the authentication code and the state
+  parameter sent in from the /login endpoint. The cellxgene server then exchanges the
+  authentication code for an ID Token. Once complete, the cellxgene server redirects the client
+  back to the dataset, and directs the browser to store the ID Token in a cookie.
+
```

Comment:
I remember that the health/ API endoint ended up changing as well. Can you remark here (or maybe elsewhere is appropriate) why this change was needed and how it changed?